File: frontend/app/src/main/java/com/example/easypark/easyparkfinal/network/UsuarioService.java
package com.example.easypark.easyparkfinal.network;

import com.example.easypark.easyparkfinal.beans.LoginDTO;
import com.example.easypark.easyparkfinal.beans.TokenDTO;
import com.example.easypark.easyparkfinal.utils.Constants;

import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class UsuarioService {
    public static Call<TokenDTO> logarUsuario(LoginDTO login){
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(Constants.BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
        IBackendAPI service = retrofit.create(IBackendAPI.class);
        return service.logarUsuario(login);
    }
}

File: MobileTruck/app/src/main/java/easypark/com/mobiletruck/network/UsuarioService.kt
package easypark.com.mobiletruck.network


import easypark.com.mobiletruck.model.LoginDTO
import easypark.com.mobiletruck.model.Token
import easypark.com.mobiletruck.session.BackendRetrofitBuilder


import retrofit2.Call



class UsuarioService {

    fun logarUsuario(login: LoginDTO): Call<Token> {
        var instance = BackendRetrofitBuilder.instance.backendAPI
        return instance.logarUsuario(login)
        /*
        val retrofit = Retrofit.Builder()
                .baseUrl(Constants.BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
        val service = retrofit.create(BackendAPI::class.java)
        return service.logarUsuario(login)
        */
    }

    companion object {
        fun newInstance() = UsuarioService()
    }
}